plugins/modules/ndo_fabric_span_session_source.py

```diff
@@ -0,0 +1,752 @@
+    uuid = access_path_config.get("uuid")
+    if not uuid:
+        fabric_template = mso_templates.get_template("fabric_resource", access_path_config.get("template"), access_path_config.get("template_id"))
+        uuid = fabric_template.get_template_policy_uuid("fabric_resource", access_path_config.get("name"), resource_type)
+
+    return (
+        dict(vpcComponentPc=[dict(vpc=uuid, node=str(access_path_config.get("node")))]) if access_path_type == "vpc_component_pc" else {resource_type: [uuid]}
+    )
+
+
+def update_access_paths(mso, site_id, access_paths, mso_templates):
+    updated_paths = []
+    for access_path in access_paths:
+        if access_path.get("access_path_type") == "port":
+            port_uuid = access_path.get("uuid")
+            if not port_uuid:
+                port_uuid = mso.get_site_interface_details(
+                    site_id=site_id,
+                    uuid=None,
+                    node=access_path.get("node"),
+                    port=access_path.get("interface"),
+                ).get("uuid")
+            updated_paths.append(dict(accessInterfaces=[port_uuid]))
+
+        elif access_path.get("access_path_type") == "port_channel":
+            updated_paths.append(get_access_path_payload(mso_templates, access_path, "port_channel", "portChannels"))
+
+        elif access_path.get("access_path_type") == "virtual_port_channel":
+            updated_paths.append(get_access_path_payload(mso_templates, access_path, "virtual_port_channel", "virtualPortChannels"))
+
+        elif access_path.get("access_path_type") == "vpc_component_pc":
+            updated_paths.append(get_access_path_payload(mso_templates, access_path, "vpc_component_pc", "virtualPortChannels"))
+
+    if len(set([str(path) for path in updated_paths])) == len(updated_paths):
+        return updated_paths
+    else:
+        mso.fail_json(msg="Remove duplicate entries from the access_paths: {0}".format(access_paths))
+
+
+def set_fabric_span_session_source_object_details(mso_template, site_id, source):
+    if source:
+        for access_path in source.get("accessPaths", []):  # Adding the object reference name to use the update_config_with_template_and_references function
+            if access_path.get("accessInterfaces") and isinstance(access_path.get("accessInterfaces")[0], str):
+                access_path.get("accessInterfaces")[0] = mso_template.mso.get_site_interface_details(site_id, access_path.get("accessInterfaces")[0])
+            elif access_path.get("portChannels") and isinstance(access_path.get("portChannels")[0], str):
+                access_path.get("portChannels")[0] = dict(portChannel=access_path.get("portChannels")[0])
+            elif access_path.get("virtualPortChannels") and isinstance(access_path.get("virtualPortChannels")[0], str):
+                access_path.get("virtualPortChannels")[0] = dict(virtualPortChannel=access_path.get("virtualPortChannels")[0])
```

Comment:
could this index lead to issues? should we check the list is not empty?

Reply:
Yes, In the if condition I am checking two things - `access_path.get("virtualPortChannels"), isinstance(access_path.get("virtualPortChannels")[0], str)`.

The first part of the condition will take care of the empty list and dict, so we can ignore the additional check for now.